transactions/transaction_service.py

```
@@ -35,7 +35,63 @@ def __init__(self, db: sqlite3.Connection) -> None:
         :param: db: Database connection object.
         """
         self.db = db
-
+    
+    def get_transactions_for_period(self, start_day: str, end_day:str) -> list:
+        """
+        Retrieve transactions in between 2 dates.
+        :params: 
+            start_day: The first day of the retrieved transactions.
+            end_day: The last day of the retrieved transactions.
+        :returns: 
+            list: A list of transactions.
+        """
+        insert_sql = """
+            SELECT 
+                t.transaction_id,
+                t.transaction_date,
+                t.description,
+                t.money_out,
+                t.money_in,
+                t.balance,
+                a.account_number,
+                c.name
+            FROM 
+                transactions t
+            JOIN 
+                accounts a ON t.account_id = a.account_id
+            JOIN 
+                categories c ON t.category_id = c.category_id
+            WHERE t.transaction_date 
+            BETWEEN ? AND ?;
+        """
+        try:
+            cursor = self.db.connection.cursor()
+            cursor.execute(insert_sql, [start_day, end_day])
+            rows = cursor.fetchall()
+            return rows
+        except Error as e:
+            print(e)
```

raise an error instead of printing

---

Updated